### PR TITLE
Add Python RPC tutorial

### PR DESCRIPTION
**Pull request by pinheadmz**
Lemme know what you think Christopher! I'm happy to contribute to this project.

### CODE REVIEW

**Comment by ChristopherA** [comment]
Like it! @shannona, do you want to do an review or just accept?

File: 12_6_Accessing_Bitcoind_with_Python.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# 12.6: Accessing Bitcoind with Python
+
+> **NOTE:** This is a draft in progress, so that I can get some feedback from early reviewers. It is not yet ready for learning.
+
+Though command-line `curl` is the easiest way to access the `bitcoind` directly, there are many other options for doing so and most of them support more fully featured programming languages. One package for doing so in Python is [Bitcoin authproxy](https://github.com/bitcoin/bitcoin/blob/master/test/functional/test_framework/authproxy.py). It was originally [written by Jeff Garzik](https://github.com/jgarzik/python-bitcoinrpc) but is now maintained by Bitcoin Core as the primary interface for the QA testing suite. It uses an http client module to access the `bitcoind` RPC server.
+
+## Set Up Bitcoin AuthProxy
+
+The latest Bitcoin Core version of `authproxy` requires Python 3, but you can still use jgarzik's original script to run in Python 2. Download the script to your working directory, start Python, and import the module. This example uses Python 3 running on Linux:
```
Comment (shannona) [inline 12_6_Accessing_Bitcoind_with_Python.md]: If possible, we're trying to include the complete instructions for how to use these Bitcoin options, which means here we'd like to show the complete methodology for getting the Python installed and setup. Ideally, use the same Debian setup that we use for most of our examples.

File: 12_6_Accessing_Bitcoind_with_Python.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+```
+$ python3
+>>> from authproxy import AuthServiceProxy, JSONRPCException
+```
+Instantiate an `AuthServiceProxy` object with the same localhost IP address, port, username and password you set in `bitcoin.conf`. For exmaple if these are the lines in `bitcoin.conf`:
```
Comment (shannona) [inline 12_6_Accessing_Bitcoind_with_Python.md]: example

File: 12_6_Accessing_Bitcoind_with_Python.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+```
+rpcuser=bitcoinrpc
+rpcpassword=1d446e2e45e96aaba6dea515871cc497
+```
+your Python command would be:
+```
+>>> rpc_connection = AuthServiceProxy("http://%s:%s@127.0.0.1:8332"%("bitcoinrpc","1d446e2e45e96aaba6dea515871cc497"))
+
```
Comment (shannona) [inline 12_6_Accessing_Bitcoind_with_Python.md]: I assume this is part of the Python code file? It'd be great to show more of the context, and we also want to have the complete code files at the end.

File: 12_6_Accessing_Bitcoind_with_Python.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+```
+
+### Make an RPC Call
+
+Now that we have our RPC connection set up via the proxy, RPC calls as easy to make as if you were just calling `bitcoin-cli` form the command line. AuthProxy returns a Python dict that may contain decimals, so they are not super easy to convert directly to JSON. We'll make a few generic calls to `bitcoind` to get some network statistics:
```
Comment (shannona) [inline 12_6_Accessing_Bitcoind_with_Python.md]: from

File: 12_6_Accessing_Bitcoind_with_Python.md (hunk 1 above)
Comment (shannona) [inline 12_6_Accessing_Bitcoind_with_Python.md]: It'd be good to have a bit of explanation on what authproxy is and what it does, for people not familiar with the concept.

File: 12_6_Accessing_Bitcoind_with_Python.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+```
+>>> rpc_connection.getmempoolinfo()
```
Comment (shannona) [inline 12_6_Accessing_Bitcoind_with_Python.md]: So we can just call rpc_connection.anyrpccommand()? If so, we should say so. If not, we should point people to where a master list is.

File: 12_6_Accessing_Bitcoind_with_Python.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+{'usage': 42593984, 'bytes': 21357852, 'size': 17809, 'mempoolminfee': Decimal('0.00078966'), 'maxmempool': 50000000}
+>>> rpc_connection.getbestblockhash()
+'00000000000000000033a05c84da138f2a6779f34671082cfd4b9f586054c65c'
+```
+You can even use this interface to run wallet commands, and add paramaters to the RPC call. Here we ask `bitcoind` for a new address, then send that address back to `bitcoind` to get the associated private key. (In this example, the wallet is not encrypted!):
```
Comment (shannona) [inline 12_6_Accessing_Bitcoind_with_Python.md]: It'd be nice to take this step by step. Give it its own subsection. Give each command and explain the results.

**Comment by shannona** [comment]
@pinheadmz, I wanted to touch bases. I would love to get a bit of expansion, per my notes from just before the holidays, to flesh this out so that we can incorporate it.

**Comment by pinheadmz** [comment]
I'm still on board! Taking care of a few other projects this week, but I'll pick this up again soon

**Comment by shannona** [comment]
Is this still pending, or should I go ahead and close the PR? Thanks!

**Comment by shannona** [comment]
Thanks, @pinheadmz ; I'll go ahead and close this, and if anyone is ever interested in picking this up, just let me know.

**Pull request closed without merging**